tools/linter_lib/pyflakes.py

```diff
@@ -16,26 +16,25 @@ def check_pyflakes(options, by_lang):
     color = next(colors)
     pyflakes = subprocess.Popen(['pyflakes'] + by_lang['py'],
                                 stdout=subprocess.PIPE,
-                                stderr=subprocess.STDOUT,
-                                universal_newlines=True)
+                                stderr=subprocess.STDOUT)
 
     for ln in iter(pyflakes.stdout.readline, b''):
         if options.full or not (
-            'imported but unused' in ln or
-            'redefinition of unused' in ln or
+            b'imported but unused' in ln or
+            b'redefinition of unused' in ln or
             # Our ipython startup pythonrc file intentionally imports *
-            ("scripts/lib/pythonrc.py" in ln and
-             " import *' used; unable to detect undefined names" in ln) or
+            (b"scripts/lib/pythonrc.py" in ln and
+             b" import *' used; unable to detect undefined names" in ln) or
             # Special dev_settings.py import
-            "from .prod_settings_template import *" in ln or
-            ("settings.py" in ln and
-             ("settings import *' used; unable to detect undefined names" in ln or
-              "may be undefined, or defined from star imports" in ln)) or
-            ("zerver/tornado/ioloop_logging.py" in ln and
-             "redefinition of function 'instrument_tornado_ioloop'" in ln) or
-            ("zephyr_mirror_backend.py:" in ln and
-             "redefinition of unused 'simplejson' from line" in ln)):
+            b"from .prod_settings_template import *" in ln or
+            (b"settings.py" in ln and
+             (b"settings import *' used; unable to detect undefined names" in ln or
+              b"may be undefined, or defined from star imports" in ln)) or
+            (b"zerver/tornado/ioloop_logging.py" in ln and
+             b"redefinition of function 'instrument_tornado_ioloop'" in ln) or
+            (b"zephyr_mirror_backend.py:" in ln and
+             b"redefinition of unused 'simplejson' from line" in ln)):
 
-            print_err('pyflakes', color, ln)
+            print_err('pyflakes', color, ln.decode('utf-8'))
```

Comment:
Can we decode the line before checking the if condition? By doing so we can avoid converting exceptions to byte strings